packages/components/src/disabled/index.js

```diff
@@ -49,9 +49,7 @@ function Disabled( { className, children, ...props } ) {
 
 			if ( focusable.nodeName === 'A' ) {
 				focusable.setAttribute( 'tabindex', -1 );
-			}
-
-			if ( focusable.hasAttribute( 'tabindex' ) ) {
+			} else if ( focusable.hasAttribute( 'tabindex' ) ) {
```

Comment:
Good catch, that said, it seems like the fix here will miss things that have a tabindex but are not links. So IMO a better fix to is to remove tabindex only if its value is different than `-1`

Reply:
Actually, it seems there's an even better fix, restore the previous code and add `A` to `DISABLED_ELIGIBLE_NODE_NAMES`

Reply:
Great idea there! Committed it!